### PR TITLE
Refactor vgmanager

### PR DESCRIPTION
**Pull request by sp98**
- add Executor interface to run `os/exec` commands. Helps to write unit tests using mocks.
- Update `block device` utility to use Executor interface and `json` response for easy parsing.
- Update vgmanager controller to use Executor interface and `json` response for easy parsing

Tested changes on SNO-2:

```
$  oc get pods -n lvm-operator-system 
NAME                                  READY   STATUS    RESTARTS      AGE
controller-manager-664bc69459-89xx8   2/2     Running   0             18m
topo-test-6df5cf5457-sp692            1/1     Running   0             7m52s
topolvm-controller-58c8f86445-z2v4b   4/4     Running   0             15m
topolvm-node-lzcx7                    4/4     Running   8 (14m ago)   15m
vg-manager-mb985                      1/1     Running   0             15m

# sapillai @ localhost in ~/go/src/github.com/red-hat-storage/lvm-operator on git:refactor-vgmanager o [19:48:20] 
$ oc get pv                                             
NAME                                       CAPACITY   ACCESS MODES   RECLAIM POLICY   STATUS   CLAIM                      STORAGECLASS              REASON   AGE
pvc-3e72f055-8e1d-4bec-b3e8-e31565943978   20Gi       RWO            Delete           Bound    lvm-operator-system/pvc1   topolvm-provisioner-vg1            7m58s

```

### CODE REVIEW

**Review by nbalacha — CHANGES_REQUESTED** [review]
First round of review. I'll do one more round tomorrow.

File: pkg/vgmanager/vgmanager_controller.go [hunk 1]
```diff
@@ -177,6 +175,46 @@ func (r *VGReconciler) reconcile(ctx context.Context, req ctrl.Request) (ctrl.Re
 	return ctrl.Result{RequeueAfter: requeueAfter}, nil
 }
 
+func (r *VGReconciler) addDevicesToVG(vgName string, devices []internal.BlockDevice) error {
+	if len(devices) < 1 {
+		return fmt.Errorf("can't create vg with 0 devices")
```
[the remark this answers is not in the document]
Comment (sp98) [reply pkg/vgmanager/vgmanager_controller.go]: done.

File: pkg/vgmanager/lvm.go [hunk 2]
```diff
@@ -17,196 +17,36 @@ limitations under the License.
...
-	if err := c.Start(); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to list volume groups. %v", err)
 	}
-	out, err := io.ReadAll(stdout)
+
+	err = json.Unmarshal([]byte(output), &volumeGroupMap)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to unmarhsal volume group response. %v", err)
```
[the remark this answers is not in the document]
Comment (sp98) [reply pkg/vgmanager/lvm.go]: done

File: pkg/vgmanager/lvm.go [hunk 3]
```diff
@@ -17,196 +17,36 @@ limitations under the License.
...
-}
-
-// parseOutput calls lvm family and parses output from it.
-//
-// cmd is a command name of lvm family.
-// fields are comma separated field names.
-// args is optional arguments for lvm command.
-func parseOutput(cmd, fields string, args ...string) ([]LVInfo, error) {
+func ListVolumeGroups(exec internal.Executor) ([]VolumeGroup, error) {
+	volumeGroupMap := map[string][]map[string][]VolumeGroup{}
```
[the remark this answers is not in the document]
Comment (sp98) [reply pkg/vgmanager/lvm.go:40]: done

File: pkg/vgmanager/vgmanager_controller.go [hunk 4]
```diff
@@ -177,6 +175,46 @@ func (r *VGReconciler) reconcile(ctx context.Context, req ctrl.Request) (ctrl.Re
...
+	}
+
+	var cmd string
+	args := []string{vgName}
```
[the remark this answers is not in the document]
Comment (sp98) [reply pkg/vgmanager/vgmanager_controller.go]: done.

**Comment by nbalacha** [comment]
Please correct the commit messages (formats)

**Review by nbalacha — APPROVED** [review]

**Pull request merged**